standalone-metastore/metastore-server/src/main/resources/package.jdo

```diff
@@ -1534,10 +1534,10 @@
         <column name="RM_DUMP_EXECUTION_ID" jdbc-type="bigint" allows-null="false"/>
       </field>
       <field name="metadata">
-        <column name="RM_METADATA" jdbc-type="varchar" allows-null="true"/>
+        <column name="RM_METADATA" jdbc-type="varchar" length="4000" allows-null="true"/>
```

Comment:
Should RM_MEATADATA and RM_PROGRESS allow null? Any use case?

Reply:
Yes progress will be null initially when the entry is just made. Metadata mayn't be.

Reply:
In case when it has just added the entry with null value for metadata and progress and sink happens, 
Shouldn't this create problem while serializing metadata and progress in MetricSink.java
persistentMetric.setProgress(mapper.writeValueAsString(metric.getProgress()));
persistentMetric.setMetadata(mapper.writeValueAsString(metric.getMetadata()));

i.e is json serialization and deserialization fine in that case(values being null)?

Reply:
There are null checks present already
if (!StringUtils.isEmpty(replicationMetric.getMetadata())) {
          mReplicationMetrics.setMetadata(replicationMetric.getMetadata());
        }
        if (!StringUtils.isEmpty(replicationMetric.getProgress())) {
          mReplicationMetrics.setProgress(replicationMetric.getProgress());
        }

And json serialization handles null
Code from json serializer

 /**
     * The method to be called by {@link ObjectMapper} and {@link ObjectWriter}
     * for serializing given value, using serializers that
     * this provider has access to (via caching and/or creating new serializers
     * as need be).
     */
    public void serializeValue(JsonGenerator gen, Object value) throws IOException
    {
        _generator = gen;
        if (value == null) {
            _serializeNull(gen);
            return;
        }